aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -557,5 +557,39 @@ Tensor& NestedTensor_mul__Tensor(Tensor& self, const Tensor& other) {
       });
 }
 
+Tensor select_nested(const Tensor& self, int64_t dim, int64_t index) {
+  TORCH_CHECK(
+    dim == 0,
+    "NestedTensor can only be selected along dimension 0 ",
+    "got dimension ", dim, " instead."
+  );
+  // size_matrix.dim() can only have 2 possible values:
+  //     * 2, as expected
+  //     * 0, when the nested tensor is empty
+  Tensor sizemat = get_nested_size_tensor(self);
+  TORCH_CHECK(
+    sizemat.dim() == 2,
+    "cannot index an empty nested tensor."
+  );
+  // nested_tensor[i] = i-th original tensor
+  int64_t ntensors = sizemat.size(0);
+  TORCH_CHECK_INDEX(
+    index >= -ntensors && index < ntensors,
```

Comment:
You could use [`maybe_wrap_dim`](https://github.com/pytorch/pytorch/blob/fd6991e71436b26978011cbd49b5d40d27db0857/aten/src/ATen/WrapDimUtils.h) for access to more canonical error messages.